front/app/hooks/useHomepageSettingsFeatureFlag.test.ts

```diff
@@ -0,0 +1,86 @@
+import { renderHook } from '@testing-library/react-hooks';
+import useHomepageSettingsFeatureFlag from './useHomepageSettingsFeatureFlag';
+import useAppConfiguration from './useAppConfiguration';
+import useHomepageSettings from './usehomepageSettings';
+
+jest.mock('hooks/useHomepageSettings', () => {
+  return jest.fn(() => ({
+    data: {
+      attributes: {
+        events_widget_enabled: true,
+        top_info_section_enabled: true,
+        bottom_info_section_enabled: false,
+      },
+    },
+  }));
+});
+
+jest.mock('hooks/useAppConfiguration', () =>
+  jest.fn(() => ({
+    data: {
+      attributes: {
+        settings: { events_widget: { allowed: true } },
+      },
+    },
+  }))
+);
+
+describe('useHomepageSettingsFeatureFlag', () => {
+  it('should return true when widget is allowed and enabled', () => {
+    const { result } = renderHook(() =>
+      useHomepageSettingsFeatureFlag({
+        sectionEnabledSettingName: 'events_widget_enabled',
+        appConfigSettingName: 'events_widget',
+      })
+    );
+    expect(result.current).toBe(true);
+  });
+
+  it('should return true when checking an enabled homepageSetting without a corresponding app config section', () => {
+    const { result } = renderHook(() =>
+      useHomepageSettingsFeatureFlag({
+        sectionEnabledSettingName: 'top_info_section_enabled',
+      })
+    );
+    expect(result.current).toBe(true);
+  });
+
+  it('should return false when checking a disabled homepageSetting without a corresponding app config section', () => {
+    const { result } = renderHook(() =>
+      useHomepageSettingsFeatureFlag({
+        sectionEnabledSettingName: 'bottom_info_section_enabled',
+      })
+    );
+    expect(result.current).toBe(false);
+  });
+
+  it('should return false when widget is allowed but not enabled', () => {
+    (useHomepageSettings as jest.Mock).mockReturnValue({
+      data: { attributes: { events_widget_enabled: false } },
+    });
+    const { result } = renderHook(() =>
+      useHomepageSettingsFeatureFlag({
+        sectionEnabledSettingName: 'events_widget_enabled',
+        appConfigSettingName: 'events_widget',
+      })
+    );
+    expect(result.current).toBe(false);
+  });
+
+  it('should return false when widget is enabled but not allowed', () => {
+    (useAppConfiguration as jest.Mock).mockReturnValue({
+      data: {
+        attributes: {
+          settings: { events_widget: { allowed: false } },
+        },
+      },
+    });
+    const { result } = renderHook(() =>
+      useHomepageSettingsFeatureFlag({
+        sectionEnabledSettingName: 'events_widget_enabled',
+        appConfigSettingName: 'events_widget',
+      })
+    );
+    expect(result.current).toBe(false);
+  });
+});
```

Comment:
Great, thanks for adding these! Could we add 1 more test that tests events_widget for both enabled and allowed as false? So we cover all optioins.